### DISCUSSION TITLE
How to pass function use define hook

### DISCUSSION THREAD

**Discussion by shanjingheng** [vuepress/core#1279] [Q&A] [answered]
I'm using the docsearch plugin and found that passing function type parameters are lost like this.
```
transformItems: (items)=>{
        return items.map(items=>({ ...items, content: 'xxxxx'}))
},
```
This method was not found in the Vue component __DOCSEARCH_OPTIONS__ variable.
Maybe The value be automatically processed by JSON.stringify().
How can I made it?

**Comment by Mister-Hope** [comment]
I think features like this can be only done with somthing like `new Fucntion` and `eval`

**Comment by Mister-Hope** [comment, marked as answer]
You should set them as client options, maybe you can see how `vuepress-plugin-comment2` support this.
